### PR TITLE
Fix incorrect parsing of synchronous_standby_names for postgres 9.6.

### LINKED ISSUE #242: There is a bracket in my synchronousStandbys
From my sentinel log after using `stolonctl update --patch` to change `minSynchronousStandbys`:

```
[E] 2017-03-10T15:24:25Z sentinel.go:234: no keeper info available db=a97fd754 keeper=stolon_3
[I] 2017-03-10T15:24:25Z sentinel.go:743: master db is failed db=a97fd754 keeper=stolon_3
[I] 2017-03-10T15:24:25Z sentinel.go:754: trying to find a new master to replace failed master
[W] 2017-03-10T15:24:25Z sentinel.go:771: cannot choose synchronous standby since the latest master reported synchronous standbys are different from the db spec ones reported=([]string) (len=1 cap=4) {
 (string) (len=9) "f7b5fa8d)"
}
 spec=([]string) (len=2 cap=4) {
 (string) (len=8) "35ff791c",
 (string) (len=8) "f7b5fa8d"
}
[E] 2017-03-10T15:24:25Z sentinel.go:788: no eligible masters
```

Note `(len=9) "f7b5fa8d)"` -- there is a `)` at the end of the string!?

Is that intended?

Inspecting consul, I can also see it in the `stolon/clusterdata`:

```
... "shared_buffers":"128MB","timezone":"UTC"},"synchronousStandbys":["f7b5fa8d)"]}}
```

What's going on, is this a data corruption of the stolon state in Consul or does the bracket have a meaning?

### PR DESCRIPTION
**Pull request by nh2**
Fixes #242.

### CODE REVIEW

**Comment by sgotti** [comment]
Thanks for the PR! I'll review it tomorrow.

Would you mind adding some unit tests for the parsing function?

**Comment by nh2** [comment]
> Would you mind adding some unit tests for the parsing function?

@sgotti If you tell me where I should put them / if there's an existing framework I can add them to. In any case, I tested it with the following test cases: https://gist.github.com/nh2/8f01d33c9d8bbdc59ba73363b4a199be#file-stolon-9-6-synchronous-standby-parser-fix-test-go-L43

**Comment by sgotti** [comment]
> @sgotti If you tell me where I should put them / if there's an existing framework I can add them to. In any case, I tested it with the following test cases:

That's perfect, I adapted it to be executed as a typical go test. Currently the keeper doesn't have a test file so you should create the full file. 
https://github.com/sgotti/stolon/blob/pr245/cmd/keeper/keeper_test.go

You can run it with `go test ./cmd/keeper` or the full tests with `./test`

File: cmd/keeper/keeper.go [hunk 1]
```diff
@@ -438,6 +438,47 @@ func (p *PostgresKeeper) updatePGState(pctx context.Context) {
...
+//   standby_name [, ...]
+// two examples for this:
+//   2 (node1,node2)
+//   node1,node2
+//
+// Versions only than 9.6 only support the second format.
+//
+// This function extracts the `[node1,node2]` part for both cases.
+func parse_synchronous_standby_names(s string) ([]string, error) {
+	var spacesSplit []string = strings.Split(s, " ")
```
Comment (sgotti) [inline cmd/keeper/keeper.go]: `s/parse_synchronous_standby_names/parseSynchronousStandbyNames`
to follow standard go naming converion

File: cmd/keeper/keeper.go [hunk 2]
```diff
@@ -438,6 +438,47 @@ func (p *PostgresKeeper) updatePGState(pctx context.Context) {
...
+	} else {
+		// We don't know yet which of the 2 formats we're parsing
+		_, err := strconv.Atoi(spacesSplit[0])
+		if err == nil {
+			// We're parsing format: num_sync ( standby_name [, ...] )
+			rest := strings.Join(spacesSplit[1:], " ")
+			inBrackets := strings.TrimSpace(rest)
+			if !strings.HasPrefix(inBrackets, "(") || !strings.HasSuffix(inBrackets, ")") {
+				return nil, fmt.Errorf("parse_synchronous_standby_names format has number but lacks brackets")
+			}
```
Comment (sgotti) [inline cmd/keeper/keeper.go]: I'll change the error to something like:
`synchronous standby string has number but lacks brackets`

**Comment by sgotti** [comment]
@nh2 Sorry to bother yout but I'd like to merge this. Did you find the time to take a look at my comments and to add the unit test? Thanks!

**Comment by nh2** [comment]
@sgotti I haven't had the time yet, but I might have in the first weekend of April.

**Comment by sgotti** [comment]
@nh2 Great! Thanks.

**Comment by sgotti** [comment]
superseded by #264

**Pull request closed without merging**